Application/Interfaces/IPostService.cs

```diff
@@ -0,0 +1,17 @@
+﻿using Application.Dtos.Blog;
+using Core.Entities;
+using System.ComponentModel.DataAnnotations;
+
+namespace Application.Interfaces
+{
+    public interface IPostService
+    {
+        Task<List<FeedPostVm>> GetAllPublicPosts();
+        Task<PostDetailsVm> GetPost(Guid id);
+        Task<List<BlogPostVm>> GetPublicPostsByUser(Guid userId);
+        Task<List<PrivatePostVm>> GetPrivatePostsByUser();
+        Task Create(ExtendedCreatePostDto createPostDto);
+        Task<ValidationResult> CreateComment(CreateCommentDto createCommentDto);
```

Comment:
This is actually okay if we think of `Post` entity as a domain aggregate. However, it will be harder to add functionality to the comments (edit, like, etc) if they're so closely linked to the Post entity.